### PR TITLE
Use typedef instead of #define. Use int64_t to be unambiguous.

### PR DESCRIPTION
**Pull request by mgates3**
Same changes as https://github.com/icl-utk-edu/blaspp/pull/56. Requires that to be merged first.

Builds on top of PR #35.

### CODE REVIEW

File: test/test_gbcon.cc [hunk 1]
```diff
@@ -66,7 +65,7 @@ void test_gbcon_work( Params& params, bool run )
     int64_t info_tst = lapack::gbcon( norm, n, kl, ku, &AB[0], ldab, &ipiv_tst[0], anorm, &rcond_tst );
     time = testsweeper::get_wtime() - time;
     if (info_tst != 0) {
-        fprintf( stderr, "lapack::gbcon returned error %lld\n", (lld) info_tst );
+        fprintf( stderr, "lapack::gbcon returned error %lld\n", llong( info_tst ) );
```
Comment (ayarkhan) [inline test/test_gbcon.cc:68]: Where is the `llong` defined?

Comment (ayarkhan) [reply test/test_gbcon.cc:68]: Found in include/lapack/utils.hh.

**Review by ayarkhan — COMMENTED** [review]
Changes to llong and int64_t look fine.
Just waiting for https://github.com/icl-utk-edu/blaspp/pull/56 to be merged.
Then CI can test this.

File: test/test_gbcon.cc (hunk 1 above)
Comment (mgates3) [reply test/test_gbcon.cc:68]: It's in 2 places: for code inside the `lapack` namespace, it's in `include/lapack/utils.hh`. For the testers, which are outside the `lapack` namespace, it's also in `test/test.hh`. That avoids repeating it in every file, and having to remove it if the compiler complains that it is unused in a particular file.

**Review by ayarkhan — APPROVED** [review]

**Comment by ayarkhan** [comment]
CI is showing some errors while building on gpu_nvidia; e.g.:
`lapackpp/test/cblas_wrappers.cc:43:5: error: cannot convert 'int*' to 'const lapack_int* `

**Review by ayarkhan — APPROVED** [review]

**Pull request merged**